src/memory_segments.rs

```diff
@@ -18,13 +22,62 @@ impl PySegmentManager {
     pub fn add(&self) -> PyResult<PyRelocatable> {
         Ok(self.vm.borrow_mut().add_memory_segment().into())
     }
+
+    pub fn write_arg(
+        &self,
+        py: Python<'_>,
+        ptr: PyMaybeRelocatable,
+        arg: Py<PyAny>,
+    ) -> PyResult<PyObject> {
+        // Recursive function which inserts every integer it finds, while also inserting references
+        // to new segments when an iterable is found. Those new segments will also be filled with
+        // data using this same function.
+        fn write_arg_iter(
+            py: Python,
+            ptr: &MaybeRelocatable,
+            arg_iter: &PyIterator,
+            vm: &mut VirtualMachine,
+        ) -> PyResult<MaybeRelocatable> {
+            let mut ptr = Cow::Borrowed(ptr);
+            for value in arg_iter {
+                let value = value?;
+
+                ptr = Cow::Owned(match PyIterator::from_object(py, value) {
+                    Ok(arg_iter) => {
+                        let segment_ptr =
+                            MaybeRelocatable::RelocatableValue(vm.add_memory_segment());
+                        write_arg_iter(py, &segment_ptr, arg_iter, vm)?;
+                        vm.load_data(&ptr, vec![segment_ptr]).map_err(to_py_error)?
+                    }
+                    Err(_) => {
+                        let value = value.extract::<PyMaybeRelocatable>()?;
+                        vm.load_data(&ptr, vec![value.to_maybe_relocatable()])
```

Comment:
```suggestion
                        vm.load_data(&ptr, vec![value.into()])
```
The method to_maybe_relocatable() is no needed when there is already a From implementation for this specific conversion, it will be removed soon